client/src/rr/dnssec/signer.rs

```diff
@@ -477,25 +503,76 @@ impl Signer {
     ///  being verified.
     ///
     ///  ---
-    pub fn sign_message(&self, message: &Message, pre_sig0: &SIG) -> DnsSecResult<Vec<u8>> {
+    fn sign_message(&self, message: &Message, pre_sig0: &SIG) -> ProtoResult<Vec<u8>> {
         tbs::message_tbs(message, pre_sig0).and_then(|tbs| self.sign(&tbs))
     }
+}
 
-    /// Signs a hash.
-    ///
-    /// This will panic if the `key` is not a private key and can be used for signing.
-    ///
-    /// # Arguments
-    ///
-    /// * `hash` - the hashed resource record set, see `rrset_tbs`.
-    ///
-    /// # Return value
-    ///
-    /// The signature, ready to be stored in an `RData::RRSIG`.
-    pub fn sign(&self, tbs: &tbs::TBS) -> DnsSecResult<Vec<u8>> {
-        self.key
-            .sign(self.algorithm, tbs)
-            .map_err(|e| e.into())
+impl MessageFinalizer for Signer {
+    fn finalize_message(&self, message: &Message, current_time: u32) -> ProtoResult<Vec<Record>> {
```

Comment:
@briansmith this is what I was referring to in how I wanted to generify the concept of the MessageSigner, now MessageFinalizer. This generically implements the concept of building some additional set of records to inject into the final Messsage just prior to Send. In `client` this is statically bound to `Signer`.